fabtests/include/windows/pthread.h

```diff
@@ -0,0 +1,310 @@
+{
+	void* (*routine)(void*);
+	void* arg;
+} fi_thread_arg;
+
+static DWORD WINAPI ofi_thread_starter(void* arg)
+{
+	fi_thread_arg data = *(fi_thread_arg*)arg;
+	free(arg);
+	return (DWORD)(uint64_t)data.routine(data.arg);
+}
+
+static inline int pthread_create(pthread_t* thread, void* attr, void *(*routine)(void*), void* arg)
+{
+	(void) attr;
+	fi_thread_arg* data = (fi_thread_arg*)malloc(sizeof(*data));
+	data->routine = routine;
+	data->arg = arg;
+	DWORD threadid;
+	*thread = CreateThread(0, 0, ofi_thread_starter, data, 0, &threadid);
+	return *thread == 0;
+}
+
+static inline int pthread_equal(pthread_t t1, pthread_t t2)
+{
+	(void)t1;
+	(void)t2;
+	/*
+	 * TODO: temporary solution
+	 * Need to implement
+	 */
+	return ENOSYS;
+}
+
+static inline int pthread_cancel(pthread_t thread)
+{
+	(void)thread;
+	/*
+	 * TODO: temporary solution
+	 * Need to implement
+	 */
+	return ENOSYS;
+}
+
+static inline pthread_t pthread_self(void)
+{
+	/*
+	 * TODO: temporary solution
+	 * Need to implement
+	 */
+	return (pthread_t) ENOSYS;
+}
+
+static inline int pthread_yield(void)
+{
+	(void) SwitchToThread();
+	return 0;
+}
+
+/*
+ * TODO: temporary solution
+ * Need to re-implement
+ */
+typedef void(*pthread_cleanup_callback_t)(void *);
+typedef struct pthread_cleanup_t
+{
+	pthread_cleanup_callback_t routine;
+	void *arg;
+} pthread_cleanup_t;
+
+/* Read-Write lock implementation */
+
+typedef struct {
+    SRWLOCK	lock; /* Windows Slim Reader Writer Lock */
+    bool	write_mode;
+} pthread_rwlock_t;
+typedef void pthread_rwlockattr_t;
+
+static inline int pthread_rwlock_init(pthread_rwlock_t *rwlock, const pthread_rwlockattr_t *attr)
+{
+	(void)attr;
+	if (rwlock) {
+		InitializeSRWLock(&(rwlock->lock));
+		rwlock->write_mode = false;
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_destroy(pthread_rwlock_t *rwlock)
+{
+	/* No SRWLock cleanup function */
+	(void)rwlock;
+	return 0;
+}
+
+static inline int pthread_rwlock_rdlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock) {
+		AcquireSRWLockShared(&(rwlock->lock));
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_tryrdlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock && TryAcquireSRWLockShared(&(rwlock->lock))) {
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_wrlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock) {
+		AcquireSRWLockExclusive(&(rwlock->lock));
+		rwlock->write_mode = true;
+		return 0;
+	}
+	return 1;
+}
+
+static inline int pthread_rwlock_trywrlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock && TryAcquireSRWLockExclusive(&(rwlock->lock))) {
+		rwlock->write_mode = true;
+		return 0;
+	}
+	return 1;
+}
+
+
+static inline int pthread_rwlock_unlock(pthread_rwlock_t *rwlock)
+{
+	if (rwlock) {
+		if (rwlock->write_mode) {
+			rwlock->write_mode = false;
+			ReleaseSRWLockExclusive(&(rwlock->lock));
+		} else {
+			ReleaseSRWLockShared(&(rwlock->lock));
+		}
+		return 0;
+	}
+	return 1;
+}
+
+
+typedef struct {
+	int needed;
```

Comment:
I would change the field name to `count` to match the parameter name used in the man page.